genmod/score_variants/compound_scorer.py

```diff
@@ -83,109 +161,134 @@ def run(self):
-                        variant_dict=variant
-                    )
-                    
-                    variant = replace_vcf_info(
-                        keyword = 'Compounds',
-                        annotation = new_compound_string, 
-                        variant_dict=variant
-                    )
-                logger.debug("Putting variant in results_queue")
-                self.results_queue.put(variant)
+                    raw_compounds = variant['info_dict'].get('Compounds', None)
+                    for rank_score_type in RANK_SCORE_TYPE_NAMES:
+                        if raw_compounds:
+                            logger.debug("Scoring compound for variant %s" % variant_id)
+                            #Variable to see if we should correct the rank score
+                            correct_score = True
+                            # First we check if the rank score should be corrected:
+                            for family in variant['info_dict'].get('GeneticModels', '').split(','):
+                                for model in family.split(':')[-1].split('|'):
+                                    # If the variant follows any model more than the specified it should
+                                    # not be corrected
+                                    if model not in self.models:
+                                        correct_score = False
+
+                            logger.debug("Setting correct_score to {0}".format(correct_score))
+
+                            current_rank_score = rank_scores[rank_score_type][variant_id]
+
+                            logger.debug("Current rank score is {0}".format(current_rank_score))
+
+                            scored_compound_list = []
+                            only_low = True
+
+                            # One entry per family, splitted on ','
+                            # family_id and compounds splitted with ':'
+                            # list of compounds splitted on '|'
+
+                            #TODO Only checks first family now
+                            family_compound_entry = raw_compounds.split(',')[0]
+                            splitted_entry = family_compound_entry.split(':')
+                            compound_family_id = splitted_entry[0]
+                            compound_list = splitted_entry[-1].split('|')
+
+                            logger.debug("Checking compounds for family {0}".format(
+                                compound_family_id))
+
+                            #Loop through compounds to check if they are only low scored
+                            for compound_id in compound_list:
+                                compound_rank_score = rank_scores[rank_score_type][compound_id]
+                                if compound_rank_score > get_rank_score(rank_score_type=rank_score_type,
+                                                                        threshold=9,
```

Comment:
At some point, these thresholds and base rank score adjustments need to make it into the rank model (`.ini`) files. As they were not before, feel free to consider it to be out of scope of the PR.😸 

Your solution with correcting them for normalisation is a sensitive adaption to the current state of affairs, but it would be more satisfying and transparent to include the adjustments upfront in the model.